x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsIntegTests.java

```diff
@@ -275,6 +292,29 @@ public void testCreateAndRestoreSearchableSnapshot() throws Exception {
 
     }
 
+    private void assertShardFolders(String indexName, boolean snapshotDirectory) throws IOException {
+        final Index restoredIndex = resolveIndex(indexName);
+        final ShardId shardId = new ShardId(restoredIndex, 0);
+        for (String node : internalCluster().getNodeNames()) {
+            final NodeEnvironment service = internalCluster().getInstance(NodeEnvironment.class, node);
+            final ShardPath shardPath = ShardPath.loadShardPath(logger, service, shardId, null);
+            if (shardPath != null && Files.exists(shardPath.getDataPath())) {
+                assertEquals(snapshotDirectory, Files.notExists(shardPath.resolveIndex()));
```

Comment:
Can we assert that we actually asserted this on the right number of nodes (or at least one node)? Just to make sure this isn't vacuous.

Reply:
good idea